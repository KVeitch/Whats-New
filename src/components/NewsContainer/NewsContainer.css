.news-container {
  grid-area: news-container;
  overflow: scroll;
  height: calc(100vh - 80px);
}

::-webkit-scrollbar {
  width: 10px;
}

::-webkit-scrollbar-track,
::-webkit-scrollbar-corner {
  background:#273c75;
}