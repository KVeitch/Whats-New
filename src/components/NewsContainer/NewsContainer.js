import React from 'react';
import './NewsContainer.css'
import NewsArticle from '../NewsArticle/NewsArticle'


const NewsContainer = (news) => {
  const newsData = news.news.map((story,i) => {
    return(
      <NewsArticle key={i} 
        headline={story.headline} 
        img={story.img} 
        url={story.url} 
        description={story.description}
      />
    )
  })
  return <div className="news-container">
    <div id='news-container--anchor' ></div>
    {newsData}
  </div>
 };

export default NewsContainer;