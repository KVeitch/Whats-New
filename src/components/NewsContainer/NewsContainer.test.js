import React from 'react';
import { shallow } from 'enzyme';
import NewsContainer from './NewsContainer';


describe('NewsContainer', () =>{
  it('should match the snapshot', () => {
    const newsStorys = [
      {
      headline:'Big News Here',
      description:'News Descriptiopn here',
      url:'http://www.moretesting.com'
      },
      {
        headline:'More News Here',
        description:'Even More Description here',
        url:'http://www.evenmoretesting.com'
        }
    ]
    const wrapper = shallow(<NewsContainer
      news={newsStorys}
    />);

    expect(wrapper).toMatchSnapshot()
  });

})