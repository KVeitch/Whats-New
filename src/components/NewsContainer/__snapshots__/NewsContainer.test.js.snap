// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsContainer should match the snapshot 1`] = `
<div
  className="news-container"
>
  <div
    id="news-container--anchor"
  />
  <NewsArticle
    description="News Descriptiopn here"
    headline="Big News Here"
    key="0"
    url="http://www.moretesting.com"
  />
  <NewsArticle
    description="Even More Description here"
    headline="More News Here"
    key="1"
    url="http://www.evenmoretesting.com"
  />
</div>
`;
